github-management/default-branch-migration.md

```diff
@@ -0,0 +1,110 @@
+  - [ ] open PRs. The branch rename will re-trigger prow on _all_ PRs,
+    which will cause a huge spike in the CI load.
+  - [ ] periodic jobs. Changes to frequent periodics haven't been tested yet.
+
+- [ ] Create an issue in the [kubernetes/org] repo to track the branch
+rename for your repo.
+
+- [ ] Assign the GitHub Management team ([@kubernetes/owners]) for approval.
+
+- [ ] Once the issue has been approved, send a notice to your SIG's
+mailing list about the potential branch rename.
+
+## Changes pre-rename
+
+Make the following changes _before_ renaming the branch the `master` branch.
+
+Note: There might be additional changes required that have not been
+covered in this checklist.
+
+### Prowjobs
+
+Check if any of the prowjobs (presubmits, postsubmits or periodics)
+explicitly mention the `master` branch.
+
+- [ ] If a prowjob triggers on the `master` branch (`branches` field
+of the prowjob), add the `main` branch to the list
+(see [kubernetes/test-infra#20665] for an example).
+
+- [ ] If a prowjob mentions the `master` branch in `base_ref`,
+update it to the `main` branch. This change is disruptive so ensure
+that the branch is renamed between periodic job runs.
+
+- [ ] If a prowjob mentions `master` in its name, rename the job to
+include `main`. [`status-reconciler`] should automatically migrate the
+PR status contexts to the new job name but this has not been tested yet.
+This _may_ also impact testgrid history post-migration.
+
+- [ ] If a prowjob calls scripts or code in your repo that explicitly
+reference `master`, update all references to use `main`.
+
+### Prow config
+
+- [ ] If the [`milestone_applier` config] references the `master` branch,
+add the `main` branch to the config (see [kubernetes/test-infra#20675] for an example).
+
+- [ ] If the [`branch_protection` config] references the `master` branch,
+add the `main` branch to the config.
+
+## Rename the default branch
+
+Rename the default branch from `master` to `main` using the GitHub UI
+by following the [official instructions].
+
+## Changes post-rename
+
+After the default branch has been renamed to `main`, make the following
+changes.
+
+Note: There might be additional changes required that have not been
+covered in this checklist.
+
+### Prowjobs
+
+- [ ] If a prowjob still references the `master` branch in the `branches` field,
+remove the `master` branch (see [kubernetes/test-infra#20669] for an example).
+
+### Prow config
+
+- [ ] If the [`milestone_applier` config] references the `master` branch,
+remove it from the config.
+
+- [ ] If the [`branch_protection` config] references the `master` branch,
+remove it from the config.
+
+
+### Other
+
+- [ ] If any docs reference the `master` branch, update to `main`
+(URLs will be automatically redirected).
+
+- [ ] Ensure that CI and PR tests work fine.
+
+- [ ] Trial the local development experience with a pre-rename clone.
+
+- [ ] Send a notice about the branch rename to your SIG's mailing list.
```

Comment:
```suggestion
- [ ] Send a notice about the branch rename to your SIG's mailing list.
  Include the link to the [GitHub instructions to rename your local branch](https://docs.github.com/en/github/administering-a-repository/renaming-a-branch#updating-a-local-clone-after-a-branch-name-changes).
```

Slight change, but I think sending out the instructions on how to do the local change will help reduce questions.